utils/generic.py

```diff
@@ -952,3 +952,129 @@ def assign_population_pcs(
     pop_pc_pd = pop_pc_pd.drop(pc_cols, axis='columns')
 
     return pop_pc_pd, pop_clf
+
+
+def sites_concordance(mt1: hl.MatrixTable, mt2: hl.MatrixTable, filter_monomorphic: bool = True) -> hl.Table:
+    """
+
+    Computes sites concordance between two matrix tables. The result is identical to the sites concordance produced by hl.concordance
+    but computation is much faster.
+
+    :param MatrixTable mt1: left MT
+    :param MatrixTable mt2: right MT
+    :param bool filter_monomorphic: whether to filter or return monomorphic sites
+    :return: concordance table
+    :rtype: Table
+    """
+
+    def get_non_refs_table(mt: hl.MatrixTable, common_samples: hl.expr.DictExpression, filter_monomorphic: bool) -> hl.Table:
+        """
+        Creates a rows table with a unique annotation:
+        `nr`: A dict of sample_num -> genotype index in concordance table for all non-ref samples
+
+        :param MatrixTable mt: input MT
+        :param DictExpression common_samples: Hail expression containing a dict mapping samples to integer indices
+        :return: Table with non-ref samples annotation
+        :rtype: Table
+        """
+        ht = mt.select_rows(
+            nr=hl.dict(
+                hl.agg.collect(
+                    hl.agg.filter(common_samples.contains(mt.s) & hl.or_else(mt.GT.is_non_ref(), True),
```

Comment:
I tried running the code and got an error for this line:
```
Traceback (most recent call last):
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/scratch.py", line 182, in <module>
    try_slack(args.slack_channel, main, args)
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/pyscripts_y7p_f001.zip/gnomad_hail/utils/slack.py", line 112, in try_slack
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/pyscripts_y7p_f001.zip/gnomad_hail/utils/slack.py", line 95, in try_slack
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/scratch.py", line 114, in main
    variants = sites_concordance(array_mt, exome_mt)
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/pyscripts_y7p_f001.zip/gnomad_hail/utils/generic.py", line 1058, in sites_concordance
  File "/tmp/387ded7cecc74154a8f9e95cab4d28da/pyscripts_y7p_f001.zip/gnomad_hail/utils/generic.py", line 984, in get_non_refs_table
  File "</opt/conda/default/lib/python3.6/site-packages/decorator.py:decorator-gen-1112>", line 2, in filter
  File "/opt/conda/default/lib/python3.6/site-packages/hail/typecheck/check.py", line 584, in wrapper
    args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
  File "/opt/conda/default/lib/python3.6/site-packages/hail/typecheck/check.py", line 512, in check_all
    args_.append(checker.check(arg, name, arg_name))
  File "/opt/conda/default/lib/python3.6/site-packages/hail/expr/aggregators/aggregators.py", line 28, in check
    raise ExpressionException(f"{caller} must be placed outside of an aggregation. See "
hail.expr.expressions.base_expression.ExpressionException: filter must be placed outside of an aggregation. See https://discuss.hail.is/t/breaking-change-redesign-of-aggregator-interface/701
```

Reply:
I tried running with a fix:
```
        ht = mt.select_rows(
            nr=hl.dict(
                hl.agg.filter(common_samples.contains(mt.s) & hl.or_else(mt.GT.is_non_ref(), True),
                              hl.agg.collect(hl.tuple([common_samples[mt.s], hl.or_else(mt.GT.n_alt_alleles() + 2, 1)]))
                              )
            )
        ).rows()
```
which (I think) does the same thing, but now running into OOM issues. I might punt on this for now

Reply:
Ah yeah, this was the `filter` interface change. I don't think I had ever run this on large-scale data (just gnomAD exomes/genomes dups), and I can see how it may be mem hungry. Was worth a shot!